hedera-node/src/main/java/com/hedera/services/contracts/execution/HederaMessageCallProcessor.java

```diff
@@ -69,26 +80,25 @@ void executeHederaPrecompile(
         final Bytes output;
         if (contract instanceof HTSPrecompiledContract htsPrecompile) {
             final var costedResult = htsPrecompile.computeCosted(frame.getInputData(), frame);
-            if (frame.getState() == REVERT) {
-                return;
-            }
             output = costedResult.getValue();
             gasRequirement = costedResult.getKey();
         } else {
             output = contract.computePrecompile(frame.getInputData(), frame).getOutput();
             gasRequirement = contract.gasRequirement(frame.getInputData());
         }
         operationTracer.tracePrecompileCall(frame, gasRequirement, output);
-        if (frame.getRemainingGas() < gasRequirement) {
-            frame.decrementRemainingGas(frame.getRemainingGas());
-            frame.setExceptionalHaltReason(Optional.of(INSUFFICIENT_GAS));
-            frame.setState(EXCEPTIONAL_HALT);
-        } else if (output != null) {
-            frame.decrementRemainingGas(gasRequirement);
-            frame.setOutputData(output);
-            frame.setState(COMPLETED_SUCCESS);
-        } else {
-            frame.setState(EXCEPTIONAL_HALT);
+        if (frame.getState() != REVERT) {
```

Comment:
Wrather than wrapping this in a. if block I think it wojld be cleaner to use a "guard block" - 

```
    if (frame.getState() == REVERT) {
      return
    }
```

Or add it to the if/else chain with 
```
    if (frame.getState() == REVERT) {
      // nothing to do
    } else if (frame.getRemainingGas() < gasRequirement) {
    ...
```

Reply:
Fixed